neo/Consensus/ConsensusService.cs

```diff
@@ -34,7 +34,8 @@ public ConsensusService(NeoSystem system, Wallet wallet)
 
         private bool AddTransaction(Transaction tx, bool verify)
         {
-            if (context.Snapshot.ContainsTransaction(tx.Hash) ||
+            if (tx.Version < 2 || // Don't accept TX without PoW spam protection
```

Comment:
I think we can build this without needing a new tx version... perhaps just using Remark fields as Nonce, and leaving difficulty be the number of zeroes in tx hash itself.